src/python/pants/backend/docker/docker_build_context.py

```diff
@@ -30,19 +34,35 @@ class DockerBuildContext:
 @dataclass(frozen=True)
 class DockerBuildContextRequest:
     address: Address
-    context_root: str
+    build_upstream_images: bool = False
 
 
 @rule
 async def create_docker_build_context(request: DockerBuildContextRequest) -> DockerBuildContext:
+    # Get all targets to include in context.
     transitive_targets = await Get(TransitiveTargets, TransitiveTargetsRequest([request.address]))
 
-    # Get all sources, going into the build context.
+    # Get the Dockerfile from the root target.
+    dockerfiles_request = Get(
+        SourceFiles,
+        SourceFilesRequest(
+            sources_fields=[t.get(Sources) for t in transitive_targets.roots],
+            for_sources_types=(DockerImageSources,),
+        ),
+    )
+
+    # Get all dependencies for the root target.
+    root_dependencies = await MultiGet(
```

Comment:
Ah I see, this will only take the direct dependencies. That's probably OK? It means that if there is some intermediate target (e.g., of type `target()` - which is a generic target useful for aggregating dependencies - we won't traverse it, but that's probably fine for now.

Reply:
Ah, yes that's true.